include/drivers/sensor.h

```diff
@@ -666,11 +666,24 @@ static inline double sensor_value_to_double(const struct sensor_value *val)
  *
  * @param val A pointer to a sensor_value struct.
  * @param inp The converted value.
+ * @return 0 if successful, negative errno code if failure.
  */
-static inline void sensor_value_from_double(struct sensor_value *val, double inp)
+static inline int sensor_value_from_double(struct sensor_value *val, double inp)
 {
-	val->val1 = (int32_t) inp;
-	val->val2 = (int32_t)(inp * 1000000) % 1000000;
+	if (inp < INT32_MIN || inp > INT32_MAX) {
+		return -ERANGE;
+	}
+
+	double val2 = (inp - (int32_t)inp) * 1000000.0;
+
+	if (val2 < INT32_MIN || val2 > INT32_MAX) {
```

Comment:
I wonder if the `val2` (same goes for the `inp`) above is treated as an `int` or a `double` by the compiler in the comparison here?